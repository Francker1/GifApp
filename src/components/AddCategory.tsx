
import React, { useState } from 'react';
import PropTypes from 'prop-types';

type FormElement = React.FormEvent<HTMLFormElement>;

interface IMyProps {
    setCategories:any,
}


export const AddCategory:React.FC<IMyProps> = ( { setCategories }) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e:any) => {

        setInputValue(e.target.value)
    }

    const handleSubmit = (e:FormElement) => {

        e.preventDefault();

        if( inputValue.trim().length > 2 ){

            //si quisiera acumular debajo los elementos buscados:
            // setCategories( (cats:any) => [ inputValue, ...cats, ] );
            setCategories( () => [ inputValue ] );
            setInputValue("");
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
