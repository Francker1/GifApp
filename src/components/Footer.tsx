import React from 'react';


export const Footer = () => {
    return (
        <div>
            <hr />
            <p>Copyright &copy; √çtalo Franco</p>
        </div>
    )
}

