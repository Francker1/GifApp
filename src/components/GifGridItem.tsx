import React from 'react';
import PropTypes from 'prop-types';

interface IMyProps {
    props?: any,
    id: string,
    title: string,
    url: string,
    className: string
}

export const GifGridItem:React.FC<IMyProps> = ({ id, title, url }) => {
    
    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
            <p>ID: {id}</p>
        </div>
    )
}


GifGridItem.propTypes = {

    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}