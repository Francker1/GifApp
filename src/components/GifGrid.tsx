
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

interface IMyProps {
    category:string,
    className:string,
    data:[],
    loading:boolean,
}

export const GifGrid:React.FC<IMyProps> = ( {category} ) => {

    // const [images, setImages] = useState([]);

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            { loading && <p>Loading...</p> }
            
            <div className="card-grid">
                
                {
                    images.map( (img:any) => (                    
                        <GifGridItem 
                            key={img.id}
                            { ...img }
                        />
                    ))
                }
                
            </div>
        </>
    )
}
