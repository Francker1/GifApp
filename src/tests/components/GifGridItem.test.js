import React from 'react';
import {shallow} from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe("Tests in <GifGridItems />", () => {

    const id = "789ERER99FC";
    const title = "Un t√≠tulo";
    const url = "https://localhost/algo.jpg";
    const wrapper = shallow( <GifGridItem id={id} title={title} url={url} /> );
    
    test("should render the component correctly", () => {

        expect(wrapper).toMatchSnapshot();
    });

    test("should have a paragraph with a title", () => {

      //at() recoge el nodo de un elemento dado que coincida con el atributo find: ref-> https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/at.html
      const paragraph = wrapper.find("p").at(0);
      expect(paragraph.text().trim()).toBe(title);
    });

    test("should have imagen equals to url and alt of props", () => {

        const img = wrapper.find("img");
        // console.log(img.props().src)

        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    });

    test("should have className = animate__zoomIn", () => {

        const div = wrapper.exists("div.animate__zoomIn");
        // console.log(div); //true

        expect(div).toEqual(true);

        
    });
});
