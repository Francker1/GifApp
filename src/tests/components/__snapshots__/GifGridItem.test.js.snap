// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItems /> should render the component correctly 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
  <p>
    ID: 
    789ERER99FC
  </p>
</div>
`;
